ts/test/session/sending/MessageQueue_test.ts

```diff
@@ -0,0 +1,393 @@
+      const message = TestUtils.generateChatMessage();
+
+      const promise = messageQueueStub.sendMessageToDevices(devices, message);
+      await expect(promise).to.be.fulfilled;
+    });
+
+    it('can send sync message and confirm canSync is valid', async () => {
+      const devices = TestUtils.generateMemberList(3);
+      const message = TestUtils.generateChatMessage();
+      const ourDevices = [...pairedDevices, ourNumber].sort();
+
+      const promise = messageQueueStub.sendMessageToDevices(devices, message);
+      expect(promise).to.be.fulfilled;
+
+      // Check sendSyncMessage parameters
+      await tick();
+      const previousArgs = sendSyncMessageSpy.lastCall.args as [
+        ChatMessage,
+        Array<PubKey>
+      ];
+      expect(sendSyncMessageSpy.callCount).to.equal(1);
+
+      // Check that instances are equal
+      expect(previousArgs).to.have.length(2);
+
+      const argsChatMessage = previousArgs[0];
+      const argsPairedKeys = [...previousArgs[1]].map(d => d.key).sort();
+
+      expect(argsChatMessage instanceof ChatMessage).to.equal(
+        true,
+        'message passed into sendMessageToDevices was not a valid ChatMessage'
+      );
+      expect(argsChatMessage.isEqual(message)).to.equal(
+        true,
+        'message passed into sendMessageToDevices has been mutated'
+      );
+
+      argsPairedKeys.forEach((argsPaired: string, index: number) => {
+        expect(argsPaired).to.equal(ourDevices[index]);
+      });
+    });
+  });
+
+  describe('sendToGroup', () => {
+    it('can send to closed group', async () => {
+      const message = TestUtils.generateClosedGroupMessage();
+      const success = await messageQueueStub.sendToGroup(message);
+      expect(success).to.equal(true, 'sending to group failed');
+    });
+
+    it('uses correct parameters for sendToGroup with ClosedGroupMessage', async () => {
+      const message = TestUtils.generateClosedGroupMessage();
+      const success = await messageQueueStub.sendToGroup(message);
+
+      expect(success).to.equal(true, 'sending to group failed');
+
+      // Check parameters
+      await tick();
+      const previousArgs = sendMessageToDevicesSpy.lastCall.args as [
+        Array<PubKey>,
+        ClosedGroupMessage
+      ];
+      expect(previousArgs).to.have.length(2);
+
+      const argsClosedGroupMessage = previousArgs[1];
+      expect(argsClosedGroupMessage instanceof ClosedGroupMessage).to.equal(
+        true,
+        'message passed into sendMessageToDevices was not a ClosedGroupMessage'
+      );
+    });
+
+    it("won't send to invalid groupId", async () => {
+      const message = TestUtils.generateClosedGroupMessage('invalid-group-id');
+      const success = await messageQueueStub.sendToGroup(message);
+
+      // Ensure message parameter passed into sendToGroup is as expected
+      await tick();
+      expect(sendToGroupSpy.callCount).to.equal(1);
+      expect(sendToGroupSpy.lastCall.args).to.have.length(1);
```

Comment:
not sure this line is useful. sendToGroup takes always one argument right? If callCount=1, lastCall.args can only be 1, but it's not important